components/hab/src/cli.rs

```diff
@@ -861,7 +861,16 @@ pub fn sub_sup_run() -> App<'static, 'static> {
         (@arg NO_COLOR: --("no-color") "Turn ANSI color off")
         (@arg JSON: --("json-logging") "Use structured JSON logging for the Supervisor. \
             Implies NO_COLOR")
-    )
+    );
+
+    if feat::is_enabled(feat::HideHTTPSecrets) {
+        sub = sub.arg(
+            Arg::with_name("HIDE_HTTP_SECRETS")
+                .help("hides cfg section from http gateway response [default: false]")
+                .long("hide-http-secrets"),
```

Comment:
This is a bit of a nitpick but I wonder if this should be called `http-hide-cfg` or something similar.  Technically you're not hiding secrets - you're hiding the entire config, whether that config contains secrets or not.

Reply:
That comment applies not just to this line, but throughout this change.  I'm not sure if that should hold up merging or not, but it occurred to me.

Reply:
Although I guess you're also hiding the `butterfly` and `census` HTTP endpoints, in addition to the config.  There's an awful lot of hiding going on.  Maybe `http-hide-all-the-things-lol`.

Reply:
I agree. The "secret" is a matter of interpretation, not fundamental to the structure.